src/lambda/wizards/samInitWizard.ts

```diff
@@ -188,12 +406,16 @@ export class CreateNewSamAppWizard extends MultiStepWizard<CreateNewSamAppWizard
     }
 
     protected getResult(): CreateNewSamAppWizardResponse | undefined {
-        if (!this.runtime || !this.location || !this.name) {
+        if (!this.runtime || !this.template || !this.location || !this.name) {
             return undefined
         }
 
         return {
             runtime: this.runtime,
+            template: this.template,
+            region: this.region,
+            registryName: this.registryName,
+            schemaName: this.schemaName,
             location: this.location,
             name: this.name
```

Comment:
Is there a reason why all of these flags aren't included in the above if statement?

Reply:
So, the reason is region, registryName and schemaName are all optional steps and only displayed in the wizard if the user selects dynamic eventBridge template. The fact that we are checking the name, kind of seals the deal that user has gone through all the steps. This could be problematic though if the order of these steps modified. I can add an extra if statement for eventBridgeStarterAppTemplate and make sure that those optional values are selected .